File: frontend/src/app/model/userDTO.ts
import { Address } from "./address";

export class UserDTO{
    public name!: string;
    public surname!: string;
    public password!: string;
    public address!: Address;
    public jmbg!: string;
    public email!: string;
    public occupation!: string;
    public gender!: string;

    public constructor(obj?: any){
      if(obj){
        this.name = obj.name;
        this.surname = obj.surname;
        this.password = obj.password;
        this.address = obj.address;
        this.jmbg = obj.jmbg;
        this.occupation = obj.occupation;
        this.gender = obj.gender;
        this.email = obj.email;
      }
    }
}

File: backend/src/main/java/group8/bloodbank/model/DTO/UserDTO.java
package group8.bloodbank.model.DTO;

import com.fasterxml.jackson.annotation.JsonProperty;
import group8.bloodbank.model.Address;
import group8.bloodbank.model.Gender;
import lombok.Data;

@Data
public class UserDTO {
    private long id;
    private String name;
    private String surname;
    private String email;
    private String password;
    @JsonProperty
    private Address address;
    private String jmbg;
    private String occupation;
    private Gender gender;
}
